### PR TITLE
Show me the love Preact 💖

### PR DESCRIPTION
**Pull request by awinogradov**
#145

### CODE REVIEW

**Comment by Yeti-or** [comment]
But we want `(Preact || React).love();`
not only one.
We need to rewrite code or generate it so it could work with both technologies.
smth as ( bem-react-core || bem-preact-core) with one sourcecode, but with different npm modules and tests covering both worlds.

**Comment by awinogradov** [comment]
@Yeti-or It's example of how to ;)

File: package.json [hunk 1]
```diff
@@ -16,32 +16,30 @@
...
-    "react-addons-test-utils": "^15.5.1",
-    "react-test-renderer": "^15.5.4"
+    "preact": "^8.2.5",
+    "proptypes": "^1.1.0"
   },
   "peerDependencies": {
-    "react": "^15.0.0-0 || ^16.0.0-0",
-    "react-dom": "^15.0.0-0 || ^16.0.0-0"
+    "preact": "^8.2.5",
+    "proptypes": "^1.1.0"
```
Comment (zxqfox) [inline package.json]: In the proptypes readme I see:

> This is literally copied from the React 0.14 codebase

wtf? preact doesn't work properly with prop-types module?

File: src/renderTag.js [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
-        const className = addBemClassName ? classNameBuilder.stringify(
-            block,
-            elem,
-            mods,
-            mixes,
-            cls
-        ) : undefined;
+        const className = addBemClassName?
+            classNameBuilder.stringify(block, elem, mods, mixes, cls) :
+            undefined;
```
Comment (zxqfox) [inline src/renderTag.js]: there are no changes, right?

File: tests/bemClassName.spec.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import React from 'react';
-import { shallow, mount } from 'enzyme';
+import { h } from 'preact';
+import { deep } from 'preact-render-spy';
```
Comment (zxqfox) [inline tests/bemClassName.spec.js]: What about making an abstraction layer to cover both of them?
Looks like bemzyne isn't so dead, huh?

File: tests/content.spec.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 describe('Entity with declaration', () => {
     it('Entity should have passed content', () => {
-        expect(render(<SimpleBlock>content</SimpleBlock>).text())
+        expect(shallow(<SimpleBlock>content</SimpleBlock>).text())
             .toBe('content');
     });
 
     it('Entity should have declared content', () => {
-        expect(render(<MyBlock>extend</MyBlock>).text())
+        expect(shallow(<MyBlock>extend</MyBlock>).text())
```
Comment (zxqfox) [inline tests/content.spec.js]: finally render doesn't mean shallow

File: package.json (hunk 1 above)
Comment (Yeti-or) [reply package.json]: nope, see docs: https://preactjs.com/guide/differences-to-react#whats-missing

Comment (zxqfox) [reply package.json]: Holy hell!

**Pull request closed without merging**